src/NHibernate/Logging.cs

```diff
@@ -101,135 +138,183 @@ private static string GetNhibernateLoggerClass()
 			return nhibernateLoggerClass;
 		}
 
+		[Obsolete("Implement INHibernateLoggerFactory instead")]
 		public static void SetLoggersFactory(ILoggerFactory loggerFactory)
 		{
-			instance = new LoggerProvider(loggerFactory);
-		}
-
-		private LoggerProvider(ILoggerFactory loggerFactory)
-		{
-			this.loggerFactory = loggerFactory;
-		}
-
-		public static IInternalLogger LoggerFor(string keyName)
-		{
-			return instance.loggerFactory.LoggerFor(keyName);
-		}
-
-		public static IInternalLogger LoggerFor(System.Type type)
-		{
-			return instance.loggerFactory.LoggerFor(type);
-		}
-	}
-
-	public class NoLoggingLoggerFactory: ILoggerFactory
-	{
-		private static readonly IInternalLogger Nologging = new NoLoggingInternalLogger();
-		public IInternalLogger LoggerFor(string keyName)
-		{
-			return Nologging;
-		}
-
-		public IInternalLogger LoggerFor(System.Type type)
-		{
-			return Nologging;
-		}
-	}
-
-	public class NoLoggingInternalLogger: IInternalLogger
-	{
-		public bool IsErrorEnabled
-		{
-			get { return false;}
+			_instance = new LoggerProvider(new LegacyLoggerFactoryAdaptor(loggerFactory));
 		}
 
-		public bool IsFatalEnabled
+		public static void SetLoggersFactory(INHibernateLoggerFactory loggerFactory)
 		{
-			get { return false; }
+			_instance = new LoggerProvider(loggerFactory);
 		}
 
-		public bool IsDebugEnabled
+		private LoggerProvider(INHibernateLoggerFactory loggerFactory)
 		{
-			get { return false; }
+			this._loggerFactory = loggerFactory;
 		}
 
-		public bool IsInfoEnabled
+		public static IInternalLogger2 LoggerFor(string keyName)
```

Comment:
This is a breaking change, it will breaks dependent packages like NHibernate.Caches which use LoggerProvider.LoggerFor.

If keeping this as is, it will mandate a 6.0.

For it to be 5.1, we would need to either have `IInternalLogger2` extend `IInternalLogger` (confusing, bloating) or to keep `LoggerProvider.LoggerFor` yielding `IInternalLogger`, adding for migrated logging either a `LoggerFor2` method or a `LoggerProvider2`. And of course, for keeping non migrated logging working, we would need to change `LoggerFor` for it to use an `IInternalLogger` adapter over an `IInternalLogger2` logger.

Reply:
I didn't know about NHibernate.Caches using `LoggerProvider.LoggerFor`.  I guess the best option would be to create a new `LoggerProvider.LoggerFor2`.  Rev'ing major versions has too long of a timeline.

Alternatively, and a better option if NHibernate is OK taking on a new external dependency, there is [Microsoft.Extensions.Logging.Abstractions](https://www.nuget.org/packages/Microsoft.Extensions.Logging.Abstractions/) and it already has adapters to many logging framework.

Reply:
Why not. Now it appears that this package is not CLS compliant, contrary to NHibernate and its current dependencies. So this would be another breaking change, going non CLS compliant.